src/model/page-config-resolver-test.js

```diff
@@ -268,7 +272,337 @@ describes.realWin('PageConfigResolver', {}, env => {
+          </div>';
+      addMicrodata(divElement);
+      readyState = 'complete';
+      return new PageConfigResolver(gd).resolveConfig().then(config => {
+        expect(config.isLocked()).to.be.false;
+        expect(config.getProductId()).to.equal('pub1:premium');
+      });
+    });
+
+    it('should wait for element ready', () => {
+      const resolver = new PageConfigResolver(gd);
+      const element = createElement(doc, 'div');
+      addMicrodata(element);
+
+      // Empty content.
+      let config = resolver.check();
+      readyState = 'complete';
+      expect(config).to.be.null;
+
+      // Add content.
+      const divElement = createElement(doc, 'div');
+      divElement.innerHTML =
+          '<div itemscope itemtype="http://schema.org/NewsArticle"> \
+            <div itemprop="isPartOf" itemscope itemtype="http://schema.org/CreativeWork http://schema.org/Product"> \
+            <meta itemprop="name" content="New York Times"/> \
+            <meta itemprop="productID" content="pub1:premium"/> \
+          </div> \
+          <div itemprop="hasPart" itemscope itemtype="http://schema.org/WebPageElement"> \
+            <meta itemprop="isAccessibleForFree" content=true/> \
+            <meta itemprop="cssSelector" content=".paywalled-section"/> \
+          </div> \
+          <div itemprop="articleBody" class="paywalled-section"> \
+            Paid content possibly. \
+          </div> \
+        </div>';
+      addMicrodata(divElement);
+      config = resolver.check();
+      expect(config).to.be.ok;
+      expect(config.getProductId()).to.equal('pub1:premium');
+      return expect(resolver.resolveConfig()).to.eventually.equal(config);
+    });
+
+    it('new sibling changes page config', () => {
+      const divElement = createElement(doc, 'div');
+      divElement.innerHTML =
+          '<div itemscope itemtype="http://schema.org/NewsArticle" id="top"> \
+            <div itemprop="isPartOf" itemscope itemtype="http://schema.org/CreativeWork http://schema.org/Product"> \
+              <meta itemprop="name" content="New York Times"/> \
+              <meta itemprop="productID" content="pub1:premium"/> \
+            </div> \
+            <div itemscope itemtype="http://schema.org/Section"> \
+              <div itemprop="hasPart" itemscope itemtype="http://schema.org/WebPageElement"> \
+                <meta itemprop="isAccessibleForFree" content="False"/> \
+                <meta itemprop="cssSelector" content=".paywalled-section"/> \
+              </div> \
+            </div> \
+            <div itemprop="articleBody" class="paywalled-section"> \
+              Paid content possibly. \
+            </div> \
+          </div>';
+      addMicrodata(divElement);
+      readyState = 'complete';
+
+      const resolver = new PageConfigResolver(gd);
+      resolver.resolveConfig().then(config => {
+        expect(config.isLocked()).to.be.true;
+        expect(config.getProductId()).to.equal('pub1:premium');
```

Comment:
Since this is within a promise - there's no guarantee that these checks will actually execute within the test. To guarantee it, you can either return this promise or use `yield`.